ITs fail on Java 11

ITs fail on Java 11:

[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  38.369 s
[INFO] Finished at: 2021-09-26T19:52:55+03:00
[INFO] ------------------------------------------------------------------------
[WARNING] The requested profile "integration-test" could not be activated because it does not exist.
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project simple: Compilation failure: Compilation failure: 
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[ERROR] -> [Help 1]

With Java 11 fails to:
```
 [proguard] Initializing...
 [proguard] Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/logging/log4j/LogManager
 [proguard] 	at proguard.classfile.editor.CodeAttributeEditor.<clinit>(CodeAttributeEditor.java:113)
 [proguard] 	at proguard.classfile.util.DynamicMemberReferenceInitializer.<init>(DynamicMemberReferenceInitializer.java:87)
 [proguard] 	at proguard.Initializer.execute(Initializer.java:246)
 [proguard] 	at proguard.ProGuard.initialize(ProGuard.java:327)
 [proguard] 	at proguard.ProGuard.execute(ProGuard.java:132)
 [proguard] 	at proguard.ProGuard.main(ProGuard.java:717)
 [proguard] Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.LogManager
 [proguard] 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
 [proguard] 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
 [proguard] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
 [proguard] 	... 6 more
```
We need to start also adding some other JARs to classpath, Now we only add the ones from getProguardJars method. This is why also Kotlin obfuscation might have failed even previously.
```
		for (File p : proguardJars)
			java.createClasspath().createPathElement().setLocation(p);
```

--- a/src/it/simple/pom.xml
+++ b/src/it/simple/pom.xml
@@ -14,14 +14,10 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.1</version>
-        <configuration>
-          <source>1.6</source>
-          <target>1.6</target>
-          <compilerArgs>
-            <arg>-Xlint</arg>
-          </compilerArgs>
-        </configuration>
+        <version>3.8.0</version>
+		<configuration>
+			<release>11</release>
+		</configuration>
       </plugin>
       <plugin>
         <groupId>com.github.wvengen</groupId>
@@ -39,6 +35,14 @@
         <configuration>
           <attach>true</attach>
         </configuration>
+    <dependencies>
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-core</artifactId>
+      <version>2.14.1</version>
+      <scope>runtime</scope>
+    </dependency>
+    </dependencies>
       </plugin>
     </plugins>
   </build>

--- a/src/it/simple/proguard.conf
+++ b/src/it/simple/proguard.conf
@@ -1,5 +1,3 @@
--libraryjars <java.home>/lib/rt.jar
-
 -dontnote
 -dontwarn
 

--- a/src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java
+++ b/src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java
@@ -866,6 +866,27 @@ public class ProGuardMojo extends AbstractMojo {
 
 		}
 	}
+	
+	private Set<File> getAllPluginArtifactDependencies(ProGuardMojo mojo) throws MojoExecutionException {
+		Set<File> files = new HashSet<>(getProguardJars(mojo));
+		for (Artifact artifact : mojo.pluginArtifacts) {
+			files.add(artifact.getFile().getAbsoluteFile());
+			files.addAll(getChildArtifacts(artifact));
+		}
+		
+		return files;
+	}
+
+	private Set<File> getChildArtifacts(Artifact artifact) {
+		Set<File> files = new HashSet<>();
+		for (Object child : artifact.getDependencyTrail()) {
+			if (child instanceof Artifact) {
+				files.add(((Artifact) child).getFile().getAbsoluteFile());
+				files.addAll(getChildArtifacts((Artifact) child));
+			}
+		}
+		return files;
+	}
 
 	private List<File> getProguardJars(ProGuardMojo mojo) throws MojoExecutionException {
 
@@ -984,8 +1005,10 @@ public class ProGuardMojo extends AbstractMojo {
 		java.setTaskName("proguard");
 
 		mojo.getLog().info("proguard jar: " + proguardJars);
+		
+		Set<File> allDependencyFiles = getAllPluginArtifactDependencies(mojo);
 
-		for (File p : proguardJars)
+		for (File p : allDependencyFiles)
 			java.createClasspath().createPathElement().setLocation(p);
 		// java.createClasspath().setPath(System.getProperty("java.class.path"));
 		java.setClassname(mojo.proguardMainClass);

